library/src/main/java/com/google/android/exoplayer/extractor/ts/H262Reader.java

```diff
@@ -0,0 +1,296 @@
+                // loop advance the index by four.
+            } else if (data[i - 3] == 0 && data[i - 2] == 0 && data[i - 1] == 1) {
+                if (checkMpegVideoFrame(data[i]))
+                   return i - 3;
+            } else {
+                i -= 3;
+            }
+        }
+
+        if (prefixFlags != null) {
+            // True if the last four bytes in the data seen so far are {0,0,1, <valid MPEG2 video code>}.
+            prefixFlags[0] = length > 3
+                    ? (data[endOffset - 4] == 0 && data[endOffset - 3] == 0 && data[endOffset - 2] == 1 && checkMpegVideoFrame(data[endOffset - 1]))
+                    : length == 3 ? (prefixFlags[3] && data[endOffset - 3] == 0 && data[endOffset - 2] == 1 && checkMpegVideoFrame(data[endOffset - 1]))
+                    : length == 2 ? (prefixFlags[2] && data[endOffset - 2] == 1 && checkMpegVideoFrame(data[endOffset - 1]))
+                    : (prefixFlags[1] && checkMpegVideoFrame(data[endOffset - 1]));
+            // True if the last three bytes in the data seen so far are {0,0,1}.
+            prefixFlags[1] = length > 2 ? (data[endOffset - 3] == 0 && data[endOffset - 2] == 0 && data[endOffset - 1] == 1)
+                    : length == 2 ? (prefixFlags[2] && data[endOffset - 2] == 0 && data[endOffset - 1] == 1)
+                    : (prefixFlags[1] && data[endOffset - 1] == 1);
+            // True if the last two bytes in the data seen so far are {0,0}.
+            prefixFlags[2] = length > 1 ? data[endOffset - 2] == 0 && data[endOffset - 1] == 0
+                    : prefixFlags[3] && data[endOffset - 1] == 0;
+            // True if the last byte in the data seen so far is {0}.
+            prefixFlags[3] = data[endOffset - 1] == 0;
+         }
+
+        return endOffset;
+    }
+
+    public  boolean checkMpegVideoFrame (int data) {
+        int byte1 = data & 0xFF;
+        if (byte1 < 0x01 || byte1 > 0xaf){
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Parses the video width and height from sequence header and prepares media format
+     * @param data The buffer whose position should be advanced.
+     * @param offset The offset of the data in {@code data}.
+     * @param limit The limit (exclusive) of the data in {@code data}.
+     * @return MediaFormat object
+     */
+    private MediaFormat parseMediaFormatFromSeqHeader(byte[] data, int offset, int limit, boolean[] prefixFlags) {
+        int width;
+        int height;
+        int aspectRatioCode = 0;
+
+        // Width and Height
+        int b1 = data[4 + offset] & 0xFF;
+        int b2 = data[5 + offset] & 0xFF;
+        int b3 = data[6 + offset] & 0xFF;
+        width = (b1 << 4) | (b2 & 0xF0);
+        height = (b2 & 0x0F) << 8 | b3;
+        aspectRatioCode = (data[7 + offset] & 0xF0) >> 4;
+        float pixelWidthAspectRatio = 1; //default
+        switch(aspectRatioCode) {
+            case 2:
+                pixelWidthAspectRatio = 4 / 3;
```

Comment:
I think it should be like this, from what I have been testing with the video, because you need to calculate the pixel width scaling ratio
```java
        switch(aspectRatioCode) {
            case 2:
                pixelWidthHeightRatio = (4f / 3) / videoWidthHeightRatio;
                break;
            case 3:
                pixelWidthHeightRatio = (16f / 9) / videoWidthHeightRatio;
                break;
            case 4:
                pixelWidthHeightRatio = (1.21f / 1f) / videoWidthHeightRatio;
                break;
            default:
                // Do nothing.
                break;
        }
```